luxjs/forms/utils.js

```diff
@@ -42,26 +61,90 @@ angular.module('lux.form.utils', ['lux.services'])
                     } else {
                         name = val[nameOpts.source];
                     }
+
+                    // If the value already exists in the select, don't
+                    // add it again.
+                    for (var i=0; i<options.length; i++) {
+                        if (options[i].id === val[id]) return;
+                    }
+
                     options.push({
                         id: val[id],
                         name: name
                     });
+
+                });
+                // Sort list alphabetically by name.
+                options.sort(function(a,b) {
+                    var one = a.name.toLowerCase();
```

Comment:
Sorting can be done remotely by passing `?sortby=<field1>:<value1>:asc&sortby=<field2>:<value2>:asc` as URL parameters - that way the options will be loaded in the right order.